common/src/main/java/com/microsoft/identity/common/internal/numberMatch/NumberMatchJavaScriptInterface.kt

```diff
@@ -0,0 +1,73 @@
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.internal.numberMatch
+
+import android.webkit.JavascriptInterface
+import com.microsoft.identity.common.logging.Logger
+import com.microsoft.identity.common.nativeauth.internal.commands.GetAuthMethodsCommand
+
+/**
+ * Java Script API to be exposed as part of brokered WebView requests.
+ * Will be used by AuthUX as part of number match feature.
+ * When authenticator is installed, and phone uses MFA or PSI in an interactive flow, a number
+ * matching challenge is issued, where used is given a number and asked to open authenticator and check
+ * for the same number in authenticator UI. This feature cuts out one UI step, where this API is used to
+ * supply the number match value and store it in ephemeral storage (kept as long as current broker
+ * process is alive), where Authenticator can call a broker API to fetch the number match, and immediately
+ * prompt user for consent, rather than first asking them to check the number match.
+ */
+class NumberMatchJavaScriptInterface {
+
+    // Store number matches in a static hash map
+    // No need to persist this storage beyond the current broker process, but we need to keep them
+    // long enough for AuthApp to call the broker api to fetch the number match
+    // TODO: Is this a security concern?
+    companion object {
+        val TAG = NumberMatchJavaScriptInterface::class.java.simpleName
+        val numberMatchMap: HashMap<String?, String?> = HashMap()
+    }
+
+    /**
+     * Method to add a key:value pair of sessionID:numberMatch to static hashmap. This hashmap will be accessed
+     * by broker api to get the number match for a particular sessionID.
+     *
+     * TODO: Check, I feel params should be nullable, and should fallback to null check in AuthApp code.
```

Comment:
If you are marking them as nullable then you should also check for them being non-null before putting them in the map

Reply:
Yeah, this makes sense to me. I don't want to have be non-null to avoid a crash, so we can fall back to existing behavior. Having them nullable with a check makes sense